functions/New-DbaCustomError.ps1

```diff
@@ -0,0 +1,171 @@
+
+    .EXAMPLE
+        PS C:\> $server = Connect-DbaInstance localhost
+        PS C:\> $newMessage = New-DbaCustomError -SqlInstance $server -MessageID 70000 -Severity 1 -MessageText "test_70000"
+        PS C:\> $original = $server.UserDefinedMessages | Where-Object ID -eq 70000
+        PS C:\> $messageID = $original.ID
+        PS C:\> $severity = $original.Severity
+        PS C:\> $text = $original.Text
+        PS C:\> $language = $original.Language
+        PS C:\> $removed = Remove-DbaCustomError -SqlInstance $server -MessageID 70000
+        PS C:\> $alteredMessage = New-DbaCustomError -SqlInstance $server -MessageID $messageID -Severity $severity -MessageText $text -Language $language -WithLog
+
+        Simulates the sp_altermessage procedure which only allows the log behavior of the message to be changed.
+    #>
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'Low')]
+    param (
+        [parameter(Mandatory, ValueFromPipeline)]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [ValidateRange(50001, 2147483647)]
+        [int32]$MessageID,
+        [ValidateRange(1, 25)]
+        [int32]$Severity,
+        [ValidateLength(0, 255)]
+        [String]$MessageText,
+        [String]$Language = 'English',
+        [switch]$WithLog,
+        [switch]$Replace,
+        [switch]$EnableException
+    )
+
+    process {
+        $newMessages = @()
+
+        foreach ($instance in $SqlInstance) {
+            try {
+                $server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $SqlCredential -AzureUnsupported
+            } catch {
+                Stop-Function -Message "Error occurred while establishing connection to $instance" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+            }
+
+            $languageDetails = $server.Query("SELECT TOP 1 name, alias, msglangid FROM sys.syslanguages WHERE name = '$Language' OR alias = '$Language'")
+
+            if (Test-Bound Language) {
+                if ($null -eq $languageDetails) {
+                    Stop-Function -Message "$server does not have the $Language installed" -Target $instance -Continue
+                }
+            }
+
+            $languageName = $languageDetails.name
+            $languageAlias = $languageDetails.alias
+            $langId = $languageDetails.msglangid
+
+            if ($Pscmdlet.ShouldProcess("Creating new server message with id $MessageID on $server")) {
+                Write-Message -Level Verbose -Message "Creating new server message with id $MessageID on $server"
+                try {
+                    if (Test-Bound Replace) {
+                        $userDefinedMessage = $server.UserDefinedMessages | Where-Object { $_.ID -eq $MessageID -and ($_.Language -in $languageName, $languageAlias -or $_.LanguageID -eq $langId) } # special case for English and British English requires using the langId
```

Comment:
I'm not sure a replacement goes into a `New` command. Any reason we can't just have an `Update-DbaCustomError`?